pandas/_testing.py

```diff
@@ -956,6 +965,8 @@ def assert_numpy_array_equal(
     obj : str, default 'numpy array'
         Specify object name being compared, internally used to show appropriate
         assertion message.
+    index : numpy.ndarray, default None
```

Comment:
why are you adding this here? this by definition *only* compares numpy array objects

Reply:
In _testing.py, assert_frame_equal compares the content of the frame one column at a time, by calling assert_series_equal (1389).  This in some (most?) cases converts the series to ndarrays and calls assert_numpy_array_equal to compare them.